apollo-router-core/src/layers/deduplication.rs

```diff
@@ -128,3 +148,26 @@ where
         }
     }
 }
+
+struct Subscriber {
```

Comment:
Perhaps add a comment that instances of `Subscriber` are synchronised under the external `locked_wait_map` lock and that means we don't need to synchronise access to `self.receiver_count`. If we called `subscribe()` or `is_valid()` and we weren't holding that lock, bad things would happen...